libwget/encoding.c

```diff
@@ -277,7 +283,12 @@ const char *wget_str_to_ascii(const char *src)
 
 			if ((rc = idna_to_ascii_8z(src, &asc, IDNA_USE_STD3_ASCII_RULES)) == IDNA_SUCCESS) {
 				// debug_printf("toASCII '%s' -> '%s'\n", src, asc);
+# ifdef _WIN32
+				src = wget_strdup(asc);
+				idn_free(asc);
```

Comment:
Explanation is in idn-free.h (it is just made for Windows). Just pushed a patch to include it on _WIN32.